src/EnergyPlus/ExhaustAirSystemManager.cc

```diff
@@ -0,0 +1,966 @@
+                                                "Central Exhaust Fan Energy",
+                                                OutputProcessor::Unit::J,
+                                                thisExhSys.centralFan_Energy,
+                                                OutputProcessor::SOVTimeStepType::System,
+                                                OutputProcessor::SOVStoreType::Summed,
+                                                thisExhSys.Name);
+
+                            if (errFlag) {
+                                ShowContinueError(state, "Occurs in " + cCurrentModuleObject + " = " + thisExhSys.Name);
+                                ErrorsFound = true;
+                            }
+                        }
+                    }
+                } else {
+                    ShowSevereError(state, RoutineName + cCurrentModuleObject + "=" + thisExhSys.Name);
+                    ShowContinueError(state, "Fan Type =" + centralFanType + "not supported; ");
+                    ShowContinueError(state, "it needs to be either Fan:SystemModel or Fan:ComponentModel.");
+                    ErrorsFound = true;
+                }
+
+                thisExhSys.CentralFanIndex = centralFanIndex;
+
+                // std::string availSchName = ip->getAlphaFieldValue(objectFields, objectSchemaProps, "availability_schedule_name");
+                // int availSchNum = EnergyPlus::ScheduleManager::GetScheduleIndex(state, availSchName);
+                if (availSchNum > 0) {
+                    // normal conditions
+                } else if (availSchNum == 0) {
+                    // blank or anything like that, treat as always avaialabe
+                } else { // = -1 // not found a match
+                    availSchNum = 0;
+                    // a regular warning
+                    ShowWarningError(state, RoutineName + cCurrentModuleObject + "=" + thisExhSys.Name);
+                    ShowContinueError(state, "Could not find a match for Central Exhaust Fan's Avaiability Schedule.");
+                    ShowContinueError(state, "It will be treated as always available.");
+                }
+                thisExhSys.AvailScheduleNum = availSchNum;
+
+                // sizing
+                if (thisExhSys.SizingFlag) {
+                    SizeExhaustSystem(state, exhSysNum);
+                }
+            }
+            state.dataZoneEquip->NumExhaustAirSystems = numExhaustSystems;
+
+        } else {
+            // If no exhaust systems are defined, then do something <or nothing>:
+        }
+
+        if (ErrorsFound) {
+            ShowFatalError(state, "Errors found getting AirLoopHVAC:ExhaustSystem.  Preceding condition(s) causes termination.");
+        }
+    }
+
+    void InitExhaustAirSystem([[maybe_unused]] int &ExhaustAirSystemNum) // maybe unused
+    {
+    }
+
+    void CalcExhaustAirSystem(EnergyPlusData &state, int &ExhaustAirSystemNum, bool FirstHVACIteration)
```

Comment:
Why is `ExhaustAirSystemNum` passed by reference here?

Reply:
Thanks! This has been changed to a `const` pass by value.